### ISSUE TITLE
Move from Roundcube (+++) to SOGo

### ISSUE THREAD

**Issue by jkaberg** [mail-in-a-box/mailinabox#920]
An radicale idea, but how about moving from Roundcube/OwnCloud and others to https://sogo.nu/

If this is done, "only" thing left for mailinabox would be to harden the config's (DKIM, SPF etc.) as SOGo would handle the rest.

Demo here: http://demo.sogo.nu/SOGo/ (user: sogo1, password: sogo1)

I think this would be a nice upgrade @JoshData and remove a lot of redudant work

**Comment by Aravin** [comment]
Nice idea @jkaberg

**Comment by jkaberg** [comment]
I'm haveing a stab at this over here, https://github.com/jkaberg/mailinabox-sogo

There will be alot of changes, so I doubt @JoshData will want to merge changes back here

I'll let this issue live on to redirect people who want's to use SOGo

**Comment by jkaberg** [comment]
SOGo implemented, https://github.com/jkaberg/mailinabox-sogo/commit/6cd26113a1dc5ecbb88e76c5694e8176a81d37de

**Comment by dhpiggott** [comment]
Obviously it would be a big change relative to how mail-in-a-box currently provides the services it does, but the simplicity of having SOGo handle webmail, calendars and address books is very appealing.

I can't speak for anyone else but I would be very happy if you continue to post updates here letting us know how you get on with running/maintaining your fork.

In jkaberg@f4b4b1ef231906bdfad61065f46663ac6a5e282a (wahoo, lots of deletions, the bringers of simplicity!) I notice that you also updated things for Ubuntu 16.04. Could you say a little more about that, i.e. were any of the changes there specifically required to accommodate installing and running on 16.04?

I ask because I've got #758 on my mind; I'd quite like to update my mail-in-a-box to 16.04 to bring it in line with my others VPSen, but I've just not found the time. If it turns out you had little to no difficulty, that might motivate me (or someone else) to find time and come up with an acceptable solution to #758.

**Comment by jkaberg** [comment]
@dhpiggott Sure, I'll post updates. Current state is perfectly functional (on par with functions in this repo).

About Ubuntu 16.04: A side from some nitpicking from Ubuntu about missing config values it seems to be fine. Just review the commit you mention to get a good overview :-)

**Comment by jkaberg** [comment]
Status update: Migrated my old Mailinabox to the new SOGo solution. A few initial fixes later and everything is smooth as butter

**Comment by jkaberg** [comment]
Now a month and half later the fork is considered stable with no known issues.

**Comment by Synchro** [comment]
I was a bit surprised by this - as a webmail client, I found Sogo a big step backwards from roundcube, even the 5-year old version I'm using. It seems to be very limited, missing some basic things - like drag & drop, nested folders, message selection options, mark as read/unread/spam, and the calendar and address book functions seem like separate apps rather than anything integrated. It's also hugely wasteful of screen space, even in the sparsely populated demo. The UI animations are pretty and all, but they are also unnecessarily slow and I can see them becoming very annoying.

But that's just MHO, each to their own. Since there have been several PRs for switching to various different webmail clients (like in this PR, ), where the hard parts are already done, would it be difficult to make the webmail client pluggable? Seems like a better idea than committing/changing to anything specific, which is bound to rub some people the wrong way.

**Comment by jkaberg** [comment]
@Synchro You better try the demo again. The things you list works perfectly fine.

Also this is not an PR, it was an suggestion which now is an fork as there is no proper migration path due to technical issues.

**Comment by Synchro** [comment]
OK, I looked again and have discovered some of those do work, but they are a bit buggy.

You can't drag and drop any messages unless you have one selected, so you have to click, wait for the animation to finish, then you can drag the current or some other message. If no message is selected, or if you try to drag before the animation completes, the drag doesn't work.

If you select multiple messages, then try to drag a message that is not selected, it drags the selected ones instead, it doesn't drag the one you dragged. While I can see some logic in that, it's unlike other apps, and means that dragging one message acts differently to dragging multiple messages.

You can make nested folders, but the UI is inconsistent - the options menu for a folder doesn't appear until you select the folder, unlike the menus for the account and the inbox which are always shown (in the same way). I selected a folder, selected "Create a subfolder", it asked for a name, submitted it, but the folder did not appear. Tried again and it shows an error dialogue: 'An error occured while creating the mailbox "Sub"' - I subsequently discovered that it does not show folders when you create them, you have to click the parent folder to make it show, and it uses non-standard icons so the disclosure is not obvious. The error was because the folder already existed, but it didn't say that. Also, when you create a new folder, it closes the entire folder hierarchy, even if you're viewing one of the folders, so it's easy to lose your place in a busy mailbox. It also forgets folder open state - open a subfolder, close its parent, reopen its parent - the subfolder is now closed.

I found the mark as read/unread/spam options, but they only appear when you use the checkboxes; I can't see any way to get at them for the currently selected message - they don't appear in the list or message panes when viewing a single message, i.e. there's a difference between having a single message selected, and having a single message selected _using its checkbox_. This is unusual.

I can't find any way of selecting all read/unread, and it would be nice to see how it behaves when you have a large number of messages in a mailbox, like 30k+.

I know this isn't the place to talk about bugs in Sogo, but it really doesn't look mature enough to replace roundcube in this state.

**Comment by jkaberg** [comment]
That's your opinion and you're entitled to it, that being said you don't need to use the fork and you should redirect your comments to https://github.com/inverse-inc/sogo

**Comment by mitchellurgero** [comment]
What about rain loop? Doesn't require a database and is practically drag and drop. A simple bash script to auto configure it for domains.

**Comment by hachre** [comment]
Rainloop is just a webmail - it doesn't do the Calendar, Addressbook, Carddav, Exchange Support that SOGo does. 

I think many of the points @Synchro raised are totally valid. If we were to move to SOGo, we should also still offer Roundcube as an additional choice as well. (Essentially always install both and make them accessible through two subdirectories).

SOGo would replace _z-push_ and the ownCloud components that currently do _addressbook_, _calendar_ and _CalDAV_ & _CardDAV_. It could be a way to get away from ownCloud / nextCloud if we so desired. (We'd lose the file feature, but that was never an official mail-in-a-box feature anyway.)

That being said, if you're looking for an all-in-one solution that offers these things, you can check out https://github.com/mailcow/mailcow-dockerized. Maybe mail-in-a-box should stick with the choices it has made to offer greater variety on the "market".

**Comment by mitchellurgero** [comment]
Hello, rain loop has carddav support. I’m not sure about calendar.

However rainloop is also a nextcloud app (well it is also standalone if you so choose). It being a nextcloud app makes it painless to install.

**Comment by hachre** [comment]
I meant a CardDAV server, not client. I didn't know about the NextCloud app thing. Interesting...

**Comment by mitchellurgero** [comment]
@hachre sogo uses z-push doesn’t it? As for caldav/carddav why change that at this point? What’s the benefit?

**Comment by hachre** [comment]
No, they have their own Exchange support and unlike z-push, it works. I was just pointing out the general advantages of SOGo. It was previously discussed to maybe get away from ownCloud/NextCloud and this would enable that.

**Comment by mitchellurgero** [comment]
Oh!! I didn't know that, if it works better, than yes +1 from me to move to SOGo.

**Comment by xatage** [comment]
I'd like to see SOGo (possibly as an additional choice to Roundcube), too. First of all because of their better Exchange support and by the way Nextcloud could be left away.

**Comment by paradoxbound** [comment]
I am using Nextcloud extensively, so I am going to downvote any option to remove Nextcloud. I would be surprised if I was the only person using this feature.

**Comment by fabiofdsantos** [comment]
@paradoxbound,
Why not make nextcloud optional?

**Comment by viogq** [comment]
Sogo is OK but not stellar, neither is Rainloop.
